Sort interfaces

DL6ER opened the pull request:
# What does this implement/fix?

Sort interfaces

 - interfaces that do not depend on others (and have no children) are placed at the top of the tree view
 - interfaces that do have children get them directly assigned below them

![image](https://github.com/user-attachments/assets/4ae53c84-c066-47d6-b765-5b9bae7590c2)

---

**Related issue or feature (if applicable):** N/A

**Pull request in [docs](https://github.com/pi-hole/docs) with documentation (if applicable):** N/A

---
**By submitting this pull request, I confirm the following:** 

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against. 
2. I have commented my proposed changes within the code.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

## Checklist:

- [x] The code change is tested and works locally.
- [x] I based my code and PRs against the repositories `developmental` branch.
- [x] I [signed off](https://docs.pi-hole.net/guides/github/how-to-signoff/) all commits. Pi-hole enforces the [DCO](https://docs.pi-hole.net/guides/github/dco/) for all contributions
- [x] I [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) all my commits. Pi-hole requires signatures to verify authorship
- [x] I have read the above and my PR is ready for review.

DL6ER committed:
Sort interfaces:
 - interfaces that do not depend on others (and have no children) are placed at the top of the treeview
 - interfaces that do have children get them directly assigned below them

Signed-off-by: DL6ER <dl6er@dl6er.de>

yubiuser reviewed (commented):
I don't have a way to test this, as I don't have any virtual interfaces. But looking at your screenshot I noticed three `veth` interfaces blow `br-99`. I guess this is the master and the 3 child interfaces. To make it clearer that they belong together, could the children get a small indention?

rdwebdesign commented:
> I don't have a way to test this, as I don't have any virtual interfaces. 

Same issue here... I will try this later in a different machine.

rdwebdesign commented:
Maybe we can try using indentation and a single icon, instead of using 2 icons, similar to this image:

![image](https://github.com/user-attachments/assets/60f8c7ab-3655-459a-8a35-b86f4a4c79ec)

I will try something.

rdwebdesign commented on scripts/js/interfaces.js:
```diff
@@ -46,21 +47,32 @@ $(function () {
+      }
+
+      // Show an icon for indenting slave interfaces
+      const indentIcon = master === null ? "" : "<i class='fa fa-diagram-project fa-fw'></i> ";
```
Suggestion to use indentation + network icon:

![image](https://github.com/user-attachments/assets/0ab44914-4beb-40d1-8dbc-169a2d5ec7be)


1. use `fa-network-wired` as secondary icon for child interfaces:
```suggestion
      const indentIcon = master === null ? "" : "<i class='fa fa-network-wired fa-fw'></i> ";
```

rdwebdesign commented on scripts/js/interfaces.js:
```diff
@@ -46,21 +47,32 @@ $(function () {
-        text: interface.name + " - " + status,
+        text: indentIcon + interface.name + " - " + status,
         class: gateways.has(interface.name) ? "text-bold" : null,
         icon: "fa fa-network-wired fa-fw",
```
2. use the normal icon for master interfaces, but use an empty icon for child interfaces:
```suggestion
        icon: master === null ? "fa fa-network-wired fa-fw" : "fa fa-fw",
```

rdwebdesign commented:
Or maybe something like this:

![image](https://github.com/user-attachments/assets/f2eef818-07c1-4240-aa45-9cf7f01c23f5)

rdwebdesign committed:
Use an arrow to indicate child interfaces

- Add and format the arrow
- Remove the icon used to indent child interfaces (the arrow is enough)

Signed-off-by: RD WebDesign <github@rdwebdesign.com.br>

DL6ER commented:
I'd approve it if I could ;-)

![image](https://github.com/user-attachments/assets/54fb4a17-dfac-498c-8fc9-cac8c2a2c119)

yubiuser reviewed (changes requested):
We should squash down to 1-2 commits. Otherwise can go.

____

Before
![2025-03-08_14-12](https://github.com/user-attachments/assets/4cbc4722-83b6-4e85-ba61-bd36217586e7)

After
![2025-03-08_14-13_1](https://github.com/user-attachments/assets/6d1dd517-7dc8-4bac-81a7-9d253d020af7)

rdwebdesign commented:
> We should squash down to 1-2 commits.

I will squash my commits and FP.

DL6ER commented:
GHA seems to be having a hard time, set this to auto merge once it's done the CI stuff
